packages/common-types/src/accountHistory.ts

```diff
@@ -30,4 +31,29 @@ interface ERC20TransferEvent extends BaseEvent {
   type: TransactionEventTypes.ERC20_TRANSFER;
 }
 
+export type TransactionApiEvent = ERC20ApprovalApiEvent | ERC20TransferApiEvent;
+
+interface NetworkStructWithIcon extends Omit<NetworkStruct, 'nativeCurrency' | 'mainCurrency'> {
+  // Native token
+  nativeCurrency: TokenWithIcon;
+  // Token for network icon
+  mainCurrency: TokenWithIcon;
+}
+
+interface BaseEvent extends Omit<BaseApiEvent, 'spentInGas'> {
+  spentInGas: AmountsOfToken;
+  network: NetworkStructWithIcon;
+  explorerLink: string;
+}
+
+export interface ERC20ApprovalEvent extends BaseEvent, Omit<ERC20ApprovalApiEvent, 'token' | 'amount' | 'spentInGas'> {
+  token: TokenWithIcon;
```

Comment:
What about tokens not registered in our tokenLists? Do you have a default icon in mind, or should we have this icon field as optional?

Reply:
Yup, the TokenIcon already handles that. We should be passing TokenIcon to this. I didn't include token icon into the ui library right now since its tightly packed with the redux fo the token list